__tests__/html/activityStatusTelemetry.sendingToSendFailedToSent.html

```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html>
+<html lang="en-US">
+  <head>
+    <link href="/assets/index.css" rel="stylesheet" type="text/css" />
+    <script crossorigin="anonymous" src="/test-harness.js"></script>
+    <script crossorigin="anonymous" src="/test-page-object.js"></script>
+    <script crossorigin="anonymous" src="/__dist__/webchat-es5.js"></script>
+  </head>
+  <body>
+    <div id="webchat"></div>
+    <script>
+      async function createDirectLineForTest() {
+        const directLine = await testHelpers.createDirectLineWithTranscript([], {
+          overridePostActivity: activity => {
+            if (activity.type === 'message') {
+              setTimeout(() => {
+                directLine.activityDeferredObservable.next({
+                  ...activity,
+                  id: 'a00001',
+                  timestamp: new Date().toISOString()
+                });
+              }, 100);
+
+              return new Observable(observer => {
+                setTimeout(() => observer.next('a00001'), 800);
+              });
+            } else {
+              return workingDirectLine.postActivity(activity);
+            }
+          }
+        });
+
+        return directLine;
+      }
+
+      function activityChannelDataStateBecame(expectedState) {
+        if (typeof expectedState === 'undefined') {
+          return pageConditions.became(
+            `Activity "channelData.state" should be undefined`,
+            () => !('state' in pageObjects.getActivities()[0].channelData),
+            1000
+          );
+        }
+
+        return pageConditions.became(
+          `Activity "channelData.state" should be "${expectedState}"`,
+          () => pageObjects.getActivities()[0].channelData?.state === expectedState,
+          1000
+        );
+      }
+
+      run(async function () {
+        const telemetryEventTracking = [];
```

Comment:
I believe we want to use the following scenario to show `sending` -> `send failed` -> `sent`.

1. Send an activity
   - Expect `undefined` -> `sending`
2. Don't call `echoBack()` and `resolvePostActivity()`
3. Wait for 20s
   - Expect `sending` -> `send failed`
4. Call `resolveAll()` (which is `echoBack()` + `resolvePostActivity()`)
   - Expect `send failed` -> `sent`
   
You can use `lolex.install()` to create a fake timer, which you can advance its time manually.

Then, you can use `createDirectLineEmulator` without much code above.

I guess this should work. Let me know if it don't.

Reply:
updated